server/store.go

```diff
@@ -119,16 +119,17 @@ const (
 
 // StreamState is information about the given stream.
 type StreamState struct {
-	Msgs       uint64          `json:"messages"`
-	Bytes      uint64          `json:"bytes"`
-	FirstSeq   uint64          `json:"first_seq"`
-	FirstTime  time.Time       `json:"first_ts"`
-	LastSeq    uint64          `json:"last_seq"`
-	LastTime   time.Time       `json:"last_ts"`
-	NumDeleted int             `json:"num_deleted,omitempty"`
-	Deleted    []uint64        `json:"deleted,omitempty"`
-	Lost       *LostStreamData `json:"lost,omitempty"`
-	Consumers  int             `json:"consumer_count"`
+	Msgs       uint64            `json:"messages"`
+	Bytes      uint64            `json:"bytes"`
+	FirstSeq   uint64            `json:"first_seq"`
+	FirstTime  time.Time         `json:"first_ts"`
+	LastSeq    uint64            `json:"last_seq"`
+	LastTime   time.Time         `json:"last_ts"`
+	Subjects   map[string]uint64 `json:"subjects,omitempty"`
```

Comment:
Maybe time to revisit compressed API responses and acceptible response content negotiation like we have for server reqs.

As beore when we spoke I am not a huge fan of this approach as its inherently limited, think we'll sooner or later need to consider some streaming api responses anyway

Reply:
On KV for example - if a KV uses a nkey for the base of a subject, this effectively makes a subject 80+ chars in length. If the KV organizes several values by key, that means that for each datum, we have 2+ values, so assuming 1000 entries, we are looking at at least 2000 subjects or 160K payload for this request. With 1M entries (2M subjects) we are looking at 160MB.

Reply:
This is opt-in, by default you do not get the subject details (histogram).

We have an artificial limit of 100k subjects that if we have more than that we will error and the user can fall back to the watcher impl like we have now.

Also note that maximum payloads are not in play here since its a system generated message.

Reply:
What about just making getting subject a JS API call similar to getting consumers or stream names? This way it could be paged.

Reply:
Indeed, thats what I also suggested when this was first mooted. However paging a dynamic data set like subjects is bound to be unreliable, hence a seperate API call and streaming results would yield a better outcome.

Reply:
I considered that quite a bit, but decided to go with similar approach to deleted details for a stream info.

Reply:
We debated this quite a bit and this is what I think is best solution given all things considered.